src/main/java/codeu/controller/ChatServlet.java

```diff
@@ -144,24 +149,39 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
       return;
     }
 
+    // this code uses markdown (flexmark library) to parse to html to show on conversation chat
+    // Jsoup library is used to clean out any script and unwanted html tags
+
+    // use Jsoup to allow certain tags for parsing
+    Whitelist allowedTags = Whitelist.none(); //no tags allowed, empty whitelist
+    // now add tags to empty whitelist
+    // ins: underline, sub: subscript, strong: bold, em: italics, del: strikethrough, sup: superscript
+    allowedTags.addTags("ins", "sub", "strong", "em", "del", "sup");
+
+    // this allows for extensions to be added for markdown
+    MutableDataSet options = new MutableDataSet();
+
+    // set underline(++) and strikethrough(~~) extension for markdown
+    options.set(Parser.EXTENSIONS, Arrays.asList(InsExtension.create(), StrikethroughExtension.create()));
```

Comment:
As a good convention, keep the column length (no. of characters on each line) restricted to 100: https://google.github.io/styleguide/javaguide.html#s4.4-column-limit

Here & everywhere. This might help: https://github.com/google/google-java-format